pylsp/_utils.py

```diff
@@ -146,6 +149,19 @@ def format_docstring(contents):
     """
     contents = contents.replace('\t', u'\u00A0' * 4)
     contents = contents.replace('  ', u'\u00A0' * 2)
+
+    # If examples exist in the docstring wrap them in backticks
+    if ">>>" in contents:
+        # add an additional newline just in case the end of the
+        # docstring doesn't end in a blank line.
+        if contents[-2:] != "\n\n":
+            contents = f"{contents}\n"
+        # search for the example block regex
+        example_snippets = re.findall(EX_SNIPPET_RE, contents)
+        # wrap the snippets that were found in backticks
+        for snippet in example_snippets:
+            contents = contents.replace(snippet, f"```python\n{snippet}\n```")
```

Comment:
This will lead to unexpected results with the following snippet:

```
print(format_docstring("""
Test docstring with two examples

>>> example

>>> example 2 should work too
"""))
```

Result:

<pre>
Test docstring with two examples

```python
>>> example
```

```python
>>> example
``` 2 should work too
</pre>